pkg/automation/automation.go

```diff
@@ -71,174 +78,191 @@ func AutomateInstall() error {
-				continue
-			}
-			verifiedLicense = licenseData.License
-			license = licenseData.LicenseBytes
+	unverifiedLicense, err := kotsutil.LoadLicenseFromBytes(license)
+	if err != nil {
+		if licenseSecret.Labels != nil {
+			appSlug = licenseSecret.Labels["kots.io/app"]
 		}
+		return appSlug, errors.Wrap(err, "failed to unmarshal license data")
+	}
+	appSlug = unverifiedLicense.Spec.AppSlug
 
-		// check license expiration
-		expired, err := kotspull.LicenseIsExpired(verifiedLicense)
-		if err != nil {
-			logger.Error(errors.Wrapf(err, "failed to check if license is expired for app %s", verifiedLicense.Spec.AppSlug))
-			cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-			continue
-		}
-		if expired {
-			logger.Errorf("license is expired for app %s", verifiedLicense.Spec.AppSlug)
-			cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-			continue
-		}
+	logger.Debug("automated license install found",
+		zap.String("appSlug", appSlug))
 
-		// check if license already exists
-		existingLicense, err := kotsadmlicense.CheckIfLicenseExists(license)
-		if err != nil {
-			logger.Error(errors.Wrapf(err, "failed to check if license already exists for app %s", verifiedLicense.Spec.AppSlug))
-			cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-			continue
+	taskID := fmt.Sprintf("automated-install-slug-%s", appSlug)
+	if err := store.GetStore().SetTaskStatus(taskID, "Installing app...", AutomatedInstallRunning); err != nil {
+		logger.Error(errors.Wrap(err, "failed to set task status"))
+	}
+
+	finishedCh := make(chan struct{})
+	defer close(finishedCh)
+	go func() {
+		for {
+			select {
+			case <-time.After(time.Second):
+				if err := store.GetStore().UpdateTaskStatusTimestamp(taskID); err != nil {
+					logger.Error(errors.Wrapf(err, "failed to update task %s", taskID))
+				}
+			case <-finishedCh:
+				return
+			}
 		}
-		if existingLicense != nil {
-			resolved, err := kotslicense.ResolveExistingLicense(verifiedLicense)
-			if err != nil {
-				logger.Error(errors.Wrap(err, "failed to resolve existing license conflict"))
+	}()
+
+	defer func() {
+		if finalError == nil {
+			if err := store.GetStore().SetTaskStatus(taskID, "Install complete", AutomatedInstallSuccess); err != nil {
+				logger.Error(errors.Wrap(err, "failed to set error on install task status"))
 			}
-			if !resolved {
-				logger.Errorf("license already exists for app %s", verifiedLicense.Spec.AppSlug)
-				cleanup(&licenseSecret, verifiedLicense.Spec.AppSlug)
-				continue
+		} else {
+			if err := store.GetStore().SetTaskStatus(taskID, finalError.Error(), AutomatedInstallFailed); err != nil {
+				logger.Error(errors.Wrap(err, "failed to set error on install task status"))
 			}
 		}
+	}()
+
+	verifiedLicense, err := kotspull.VerifySignature(unverifiedLicense)
+	if err != nil {
+		return appSlug, errors.Wrap(err, "failed to verify license signature")
+	}
+	appSlug = verifiedLicense.Spec.AppSlug
```

Comment:
not needed, let's remove this line. app slug shouldn't ever change for the upstream app

Reply:
will be repushed